### PR TITLE
Improve create payment request view

### PR DESCRIPTION
**Pull request by dstrukt**
In my attempt to improve some of the create views, I've moved on Payment Requests.

Few things to note:
- Added helper text under the Email
- Added section headers
- For the future, want to improve the currency / amount based on the dropdown selector we've discussed (@dennisreimann @woutersamaey)
- For the future, want to improve and clean up the calendar / time selector (need to mock up the component), as the input field color (should be white), currently looks disabled.
- Made the back to list, a link element, consistent with the others .. for now, til deprecated.
- Left the "create" heading, until we can move to the header bar.
- For the future, want to make sure the correct store is selected, would be based on the work @woutersamaey is doing on Invoices currently.
- Added the store selector to the top, as it should be the first thing the user considers where the money is going.
- Need to improve the default spacing for the entire app to be 8px, so the spacing between form elements is 24px (3x), not the 21px .. more on the design system / CSS side of things though.

There will be a subsequent PR for this view when more changes and improvements facilitate an improved view, but for now...

Current:
Screen Shot 2021-06-27 at 10 23 41 AM

Based on my mock, minus the obvious differences:
Screen Shot 2021-06-27 at 10 29 12 AM

@dennisreimann On my fork -- feel free to make any additional changes you think might be helpful, or potentially easy based on the notes above.

As always, any feedback, suggestions welcome.

### CODE REVIEW

**Comment by woutersamaey** [comment]
With regard to the calendar / datetime picker, the argument can be made that we should not use such a fancy component. But just rely on the HTML5 native input type datetime-local, as explained here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local

Our company uses this a lot and have not encountered any issues with it. Also works great on mobile devices as the OS has its own native keyboard for it.

**Comment by dstrukt** [comment]
Maybe there was a misunderstanding, I'll try to clarify.

Agreed, that's in line with what I'm thinking too, and I like what I'm seeing in the docs.

I'm happy to leverage the three way scroll / html 5 component over our current component, was designing / iterating on what we currently had to make the level of effort minimal - no intentions to be fancy or superfluous, but what we have currently can undoubtedly be improved.

File: BTCPayServer/Views/PaymentRequest/EditPaymentRequest.cshtml [hunk 1]
```diff
@@ -18,19 +18,40 @@
...
+                            <input type="text" class="form-control" value="@Model.Stores.Single(item => item.Value == Model.StoreId).Text" readonly />
+                        }
+                        <span asp-validation-for="StoreId" class="text-danger"></span>
+                    </div>
+                    <div class="form-group">
+                        <label asp-for="Email" class="form-label"></label>
+                        <input type="email" asp-for="Email" class="form-control" />
+                        <span asp-validation-for="Email" class="text-danger"></span>
+                        <p id="PaymentRequestEmailHelpBlock" class="form-text text-muted">
+                            Recieve updates for this payment request.
```
Comment (britttttk) [inline BTCPayServer/Views/PaymentRequest/EditPaymentRequest.cshtml:45]:
```suggestion
                            Receive updates for this payment request.
```

Comment (britttttk) [reply BTCPayServer/Views/PaymentRequest/EditPaymentRequest.cshtml:45]: the worst English word

**Comment by NicolasDorier** [comment]
I think that's fine to use the current component if we are already doing it elsewhere anyway.

In a differnt PR later on, we can discuss if we could replace all of them with native one.

File: BTCPayServer/Views/PaymentRequest/EditPaymentRequest.cshtml (hunk 1 above)
Comment (dstrukt) [reply BTCPayServer/Views/PaymentRequest/EditPaymentRequest.cshtml]: Great catch @britttttk! 

It really is, and despite my best efforts... muscle memory always lets me down.

**Pull request merged**